File: buttertoast/src/main/java/com/taufik/buttertoast/ButterToast.java
package com.taufik.buttertoast;

import android.content.Context;
import android.widget.Toast;

public class ButterToast {
    private String message;
    private int resourceId;
    private Context context;

    private ButterToast(final Builder builder) {
        message = builder.message;
        resourceId = builder.resourceId;
        context = builder.context;
    }

    public ButterToast showToast(){
        Toast.makeText(context,message,Toast.LENGTH_SHORT).show();
        return this;
    }

    public static class Builder {
        private String message;
        private int resourceId;
        private Context context;

        public Builder(Context context) {
            this.context = context;
        }

        public Builder setMessage(final String message) {
            this.message = message;
            return this;
        }

        public Builder setMessage(final int resourceId) {
            this.resourceId = resourceId;
            return this;
        }

        public ButterToast create() {
            return new ButterToast(this).showToast();
        }
    }
}

File: buttertoast/src/main/java/com/taufik/buttertoast/ButterToast.kt
package com.taufik.buttertoast

import android.content.Context
import android.widget.Toast

class ButterToast private constructor(builder: Builder) {
    private val message: String? = builder.message
    private val resourceId: Int = builder.resourceId
    private val context: Context = builder.context

    fun showToast() {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
    }

    class Builder(val context: Context) {
        var message: String? = null
        var resourceId: Int = 0

        fun show()= ButterToast(this)
    }
}